packages/release-brancher/test/release-brancher.ts

```diff
@@ -0,0 +1,336 @@
+      runner = new Runner({
+        branchName: '1.x',
+        targetTag: 'v1.3.0',
+        gitHubToken: 'sometoken',
+        upstreamOwner: 'testOwner',
+        upstreamRepo: 'testRepo',
+      });
+      const newConfig = runner.updateSyncRepoSettings(config);
+      assert.ok(newConfig);
+      snapshot(newConfig);
+    });
+
+    it('updates a config with extra branches already configured', async () => {
+      const config = loadFixture('sync-repo-settings/with-extra-branches.yaml');
+      runner = new Runner({
+        branchName: '1.x',
+        targetTag: 'v1.3.0',
+        gitHubToken: 'sometoken',
+        upstreamOwner: 'testOwner',
+        upstreamRepo: 'testRepo',
+      });
+      const newConfig = runner.updateSyncRepoSettings(config);
+      assert.ok(newConfig);
+      snapshot(newConfig);
+    });
+
+    it('ignores branch if it already exists', () => {
+      const config = loadFixture('sync-repo-settings/with-extra-branches.yaml');
+      runner = new Runner({
+        branchName: '3.1.x',
+        targetTag: 'v3.1.0',
+        gitHubToken: 'sometoken',
+        upstreamOwner: 'testOwner',
+        upstreamRepo: 'testRepo',
+      });
+      const newConfig = runner.updateSyncRepoSettings(config);
+      assert.equal(newConfig, undefined);
+    });
+
+    it('ignores branch if repo has no branch protection enabled', () => {
+      const config = loadFixture('sync-repo-settings/no-branches.yaml');
+      runner = new Runner({
+        branchName: '3.1.x',
+        targetTag: 'v3.1.0',
+        gitHubToken: 'sometoken',
+        upstreamOwner: 'testOwner',
+        upstreamRepo: 'testRepo',
+      });
+      const newConfig = runner.updateSyncRepoSettings(config);
+      assert.equal(newConfig, undefined);
+    });
+  });
+
+  describe('createBranch', () => {
+    it('ignores if branch already exists', async () => {
+      runner = new Runner({
+        branchName: '1.x',
+        targetTag: 'v1.3.0',
+        gitHubToken: 'sometoken',
+        upstreamOwner: 'testOwner',
+        upstreamRepo: 'testRepo',
+      });
+      const requests = nock('https://api.github.com')
+        .get('/repos/testOwner/testRepo/git/ref/heads%2F1.x')
+        .reply(200, {ref: "refs/heads/1.x"});
+      const ref = await runner.createBranch();
+      assert.ok(ref);
+      assert.equal(ref, "refs/heads/1.x");
+      requests.done();
+    });
+
+    it('errors if cannot find SHA for tag', async () => {
+      runner = new Runner({
+        branchName: '1.x',
+        targetTag: 'v1.3.0',
+        gitHubToken: 'sometoken',
+        upstreamOwner: 'testOwner',
+        upstreamRepo: 'testRepo',
+      });
+      const requests = nock('https://api.github.com')
+        .get('/repos/testOwner/testRepo/git/ref/heads%2F1.x')
+        .reply(404)
+        .get('/repos/testOwner/testRepo/git/matching-refs/tags%2Fv1.3.0')
+        .reply(404);
+
+      let caught = false;
+      try {
```

Comment:
You can use `assert.rejects` or `assert.throws` here:

https://nodejs.org/api/assert.html#assert_assert_rejects_asyncfn_error_message

I've been trying to get a bit better about using these helpers.

Reply:
Switched to use `await assert.rejects()`